### ISSUE TITLE
Calendar items from Viggo is not imported on production

### ISSUE THREAD

**Issue by lsolesen** [vih/vih-build#26]
For some strange reason the calendar items are not imported correctly from Viggo (vih_viggo). This is really annoying.

**Comment by mglaman** [comment]
I remember this was an issue some time back. Vies was having the issue, not Vih. It was some kind of SSL issue if I recall correctly.

**Comment by mglaman** [comment]
@lsolesen I believe the URL changed, you might need to login to Viggo and see.

Here's what I get when I run _curl -i VIGGO_URL_

```
Cache-Control: private
Content-Type: text/html; charset=utf-8
Location: /Account/Login?ReturnUrl=%2FExportCalendar%2F%3FViggoId%3D87
Server: Microsoft-IIS/8.0
X-AspNetMvc-Version: 5.2
X-AspNet-Version: 4.0.30319
Set-Cookie: ASP.NET_SessionId=d5yxhfw10ks2bouwaarkpabg; path=/; HttpOnly
Set-Cookie: .ASPXAUTH=; expires=Mon, 11-Oct-1999 22:00:00 GMT; path=/; HttpOnly
X-Powered-By: ASP.NET
Strict-Transport-Security: max-age=16070400; includeSubDomains
Date: Thu, 05 Mar 2015 14:15:02 GMT
Content-Length: 177

<html><head><title>Object moved</title></head><body>
<h2>Object moved to <a href="/Account/Login?ReturnUrl=%2FExportCalendar%2F%3FViggoId%3D87">here</a>.</h2>
</body></html>

[1]-  Done                    curl -i https://vejle.viggo.dk/ExportCalendar/?ViggoId=87
[2]+  Done                    UserId=298
```

**Comment by lsolesen** [comment]
This is the link for vih.dk --> https://vejle.viggo.dk/ExportCalendar/?ViggoId=87&UserId=298&code=17bca452d0b19b39a49d3ffdc1a77faabe5ae617

When accessing it from the browser it works allright. Are you just running it from your own commandline? Why is the 'code' not menntioned ind the curl stuff?

**Comment by lsolesen** [comment]
Moved the priority for this up, as you have already been looking into this, and I am annoyed by the error and asked the whole time :)

**Comment by mglaman** [comment]
I think curl was stripping out the parameters after the ViggoId for some reason, that URL works for me from browser as well.

**Comment by mglaman** [comment]
@lsolesen ok, so I can get the iCal fine via curl and locally with the site wrapped in SSH. Database log isn't enabled in production and not sure how to get access to error log. Going to need to trigger import and watch error log (running import brings to error page on production from /import)

**Comment by mglaman** [comment]
Digging into the exception that's thrown (based off of old Drupal.org issue).

curl isn't getting an error, it's getting 0 which means it can't reach host.

```
      $data = curl_exec($download);
      if (curl_error($download)) {
        throw new HRCurlException(
          t('cURL error (@code) @error for @url', array(
            '@code' => curl_errno($download),
            '@error' => curl_error($download),
            '@url' => $url,
          )), curl_errno($download)
        );
      }
```

It's printing out the HTTP request code...

```
    if (!in_array($result->code, array(200, 201, 202, 203, 204, 205, 206))) {
      throw new Exception(t('Download of @url failed with code !code.', array('@url' => $this->url, '!code' => $result->code)));
    }
```

Vih.dk and Vies.dk are on different IPs... maybe Vih.dk got IP banned?

**Comment by lsolesen** [comment]
I do not think it works on vies.dk either?

**Comment by mglaman** [comment]
I wonder if it's an SSL issue. There is a commit/patch which allows setting SSL_VERIFYPEER to false. - http://cgit.drupalcode.org/feeds/commit/?id=7600c42. This might help solve issue.

**Comment by mglaman** [comment]
@lsolesen I tested, it's definitely not a Drupal issue. It's either the server's network, or Viggo itself

```
$ ping vejle.viggo.dk
PING vejle.viggo.dk (37.128.222.20) 56(84) bytes of data.
^C
--- vejle.viggo.dk ping statistics ---
27 packets transmitted, 0 received, 100% packet loss, time 26206ms
```

**Comment by mglaman** [comment]
@lsolesen have you heard anything from Viggo? I'm realizing my local copy is throwing exceptions because its failing to import.

**Comment by lsolesen** [comment]
They say that they are not blocking anything.
Den 25/03/2015 02.48 skrev "Matt Glaman" notifications@github.com:

**Comment by lsolesen** [comment]
This seems to be solved after recent upgrades to BOA 2.4.2.
